python/tank/authentication/sso_saml2/core/sso_saml2_core.py

```diff
@@ -564,9 +566,22 @@ def on_authentication_required(self, reply, authenticator):
         :param reply: Qt reply object. Not used.
         :param authenticator: Qt authenticator object.
         """
-        # Setting the user to an empty string tells the QAuthenticator to
-        # negociate the authentication with the user's credentials.
-        authenticator.setUser('')
+        # We take for granted that if we are on Windows, proper NTLM negociation
+        # is possible between the machine and the IdP. For other platforms, we
+        # pop an authentication dialog.
+        if platform.system() != "Windows":
```

Comment:
[nitpick] We typically use `sys.platform != "win32"` to do this. Want to go that route for consistency with the rest of sgtk?

Reply:
Sure, makes sense. I'll use `sys.platform`